nucleus/v1/types.go

```diff
@@ -0,0 +1,178 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// HubCore represents a deployment of nucleus hub core component.
+type HubCore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec represents a desired deployment configuration of nucleus hub
+	Spec HubCoreSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+	// Status represents the current status of nucleus hub
+	// +optional
+	Status HubCoreStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// HubCoreSpec represents a desired deployment configuration of nucleus hub.
+type HubCoreSpec struct {
+	// Namespace is the namespace to deploy the agent.
+	// The namespace must have a prefix of "open-cluster-management-", and if it is not set,
+	// the namespace of "open-cluster-management" is used to deploy agent.
+	// +optional
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,1,opt,name=namespace"`
```

Comment:
a hub cluster can only be a single hub because the namespaces collide.  I don't think this one needs a namespace, we'll simply choose one for it.  Sorry if I put my previous comment in teh wrong spot.